extensions/http-curl/client/HTTPStream.h

```diff
@@ -100,14 +100,14 @@ class HttpStream : public io::BaseStream {
    * @param buflen buffer to write
    *
    */
-  virtual int writeData(std::vector<uint8_t> &buf, int buflen);
+  virtual int writeData(const std::vector<uint8_t> &buf, int buflen);
```

Comment:
Same. C and C++ developers see const and they know what that means. They see the lack of const and know their data may change. If it's non const that doesn't mean it should be changed without discussion. I think these should be non const as we reserve the right to take ownership in the internal API. I would agree for external API.